contrib/mesos/pkg/offers/offers.go

```diff
@@ -441,7 +441,7 @@ func (s *offerStorage) ageOffers() {
 }
 
 func (s *offerStorage) nextListener() *offerListener {
-	obj := s.listeners.Pop()
+	obj := s.listeners.Pop(nil)
```

Comment:
not a big fan of optional arguments. why did you remove CancelablePop? That made it clear there were two different use cases.